ui/app/src/components/DateTimePicker.tsx

```diff
@@ -100,97 +120,117 @@ interface timezoneType {
-  const [ selectedTimezone, setSelectedTimezone ] = useState(timezoneObj);
+  let dateMoment;
+  let timezoneObj = timezones.find(tz => (tz.timezoneName === unescape(timeZonePickerProps.timezone)));
+  dateMoment = getDateMoment(date, timezoneObj);
+  const { formatMessage } = useIntl();
 
-  const handleDateChange = (name: string) => (date: Date | null) => {
-    let updatedDateTime = selectedDateTime;
+  const handleDateChange = (name: string) => (newDate: Date | null) => {
+    let updatedDateTime = dateMoment;
 
     switch (name) {
       case 'scheduledDate':
-        updatedDateTime.date(date.getDate());
-        updatedDateTime.month(date.getMonth());
-        updatedDateTime.year(date.getFullYear());
+        updatedDateTime.date(newDate.getDate());
+        updatedDateTime.month(newDate.getMonth());
+        updatedDateTime.year(newDate.getFullYear());
+        datePickerProps?.onDateChange?.(updatedDateTime.format(datePickerProps.dateFormat));
         break;
       case 'scheduledTime':
-        updatedDateTime.hours(date.getHours());
-        updatedDateTime.minutes(date.getMinutes());
+        updatedDateTime.hours(newDate.getHours());
+        updatedDateTime.minutes(newDate.getMinutes());
+        timePickerProps?.onTimeChange?.(updatedDateTime.format(timePickerProps.timeFormat));
         break;
     }
 
-    setSelectedDateTime(updatedDateTime);
     onChange?.(updatedDateTime);
-
-    onChangeDate?.(updatedDateTime.format(dateFormat));
-    onChangeTime?.(updatedDateTime.format(timeFormat));
   };
 
   const handleTimezoneChange = () => (event: React.ChangeEvent<{}>, timezoneObj: any) => {
     const timezone = timezoneObj.timezoneName,
-          updatedDateTime = moment.tz(selectedDateTime.format(), 'YYYY-MM-DD HH:mm A', timezone);
-    setSelectedDateTime(updatedDateTime);
-    setSelectedTimezone(timezoneObj);
+      updatedDateTime = moment.tz(dateMoment.format(), `${datePickerProps.dateFormat} ${timePickerProps.timeFormat} A`, timezone);
 
-    onChange && onChange(updatedDateTime);
-    onChangeTimezone && onChangeTimezone(timezoneObj);
+    timeZonePickerProps?.onTimezoneChange?.(timezone);
+    datePickerProps?.onDateChange?.(updatedDateTime.format(datePickerProps.dateFormat));
+    onChange?.(updatedDateTime);
   };
 
+  const formControlProps = {};
+  if (nnou(id)) {
+    formControlProps['id'] = id;
+  }
+
   return (
-    <>
+    <FormControl {...formControlProps}>
       <MuiPickersUtilsProvider utils={DateFnsUtils}>
-        { controls.includes('date') &&
-          <KeyboardDatePicker
-            format="MM/dd/yyyy"
-            margin="normal"
-            id="date-picker"
-            value={selectedDateTime.format('YYYY-MM-DD HH:mm')}
-            onChange={handleDateChange('scheduledDate')}
-            className={classes.picker}
-            InputAdornmentProps={{
+        {controls.includes('date') &&
+        <KeyboardDatePicker
+          format="MM/dd/yyyy"
+          margin="normal"
```

Comment:
What's this? Needed?

Reply:
@rart it adds space between the inputs, I can also add the styles for that spacing. Please let me know